### PR TITLE
Primitive object getters to support Null values 

### PR DESCRIPTION
**Pull request by ohadbitt**
Primitive objects throws on getters with null value so we made new getters as object

### CODE REVIEW

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 1]
```diff
@@ -231,19 +265,31 @@ public Date getDate(int i) throws SQLException {
 
     @Override
     public Time getTime(int i) throws SQLException {
-        return new Time(getDate(i).getTime());
+        LocalTime time = getLocalTime(i);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:268]: Why **local** time?

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 2]
```diff
@@ -231,19 +265,31 @@ public Date getDate(int i) throws SQLException {
...
             case "string":
             case "datetime":
+                if (get(i) == null) {
+                    return null;
+                }
                 return Timestamp.valueOf(StringUtils.chop(getString(i)).replace("T", " "));
             case "long":
             case "int":
-                return new Timestamp(getLong(i));
+                Long l = getLongObject(i);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:287]: Long.valueOf

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 3]
```diff
@@ -231,19 +265,31 @@ public Date getDate(int i) throws SQLException {
...
+                if (l == null) {
+                    return null;
+                }
 
+                return new Timestamp(l);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:292]: Why not add this as a default case?

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 4]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
...
+    public short getShortObject(String columnName) {
+        return (Short) get(columnName);
+    }
+
     @Override
     public int getInt(String columnName) {
         return (int) get(columnName);
     }
 
+    public int getInteger(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:350]: getIntegerObject

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 5]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
...
+        return (Integer) get(columnName);
+    }
+
     @Override
     public long getLong(String columnName) {
         return (long) get(columnName);
     }
 
+    public Long getLongObject(String columnName) {
+        return getLongObject(findColumn(columnName));
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:360]: We have no idea what the value type of the value will be, but the getLongObject overload is only intended for an "int".

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 6]
```diff
@@ -862,12 +958,24 @@ public Date getDate(String columnName, Calendar calendar) throws SQLException {
 
     @Override
     public Time getTime(int i, Calendar calendar) throws SQLException {
-        return new Time(getDate(i, calendar).getTime());
+        return getTime(i);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:961]: Why not involve the calendar for timezone info?

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 7]
```diff
@@ -189,11 +194,19 @@ public short getShort(int i) {
         return (short) get(i);
     }
 
+    public short getShortObject(int i) {
+        return (Short) get(i);
+    }
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:199]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 8]
```diff
@@ -203,19 +216,40 @@ public long getLong(int i) {
...
+        return (Long) obj;
+    }
+
+
     @Override
     public float getFloat(int i) {
         return (float) get(i);
     }
 
+    public Float getFloatObject(int i) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:233]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 9]
```diff
@@ -203,19 +216,40 @@ public long getLong(int i) {
...
     }
 
+    public Double getDoubleObject(int i) {
+        return (Double) get(i);
+    }
+
     @Override
     @Deprecated
     public BigDecimal getBigDecimal(int i, int i1) {
+        if (get(i) == null) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:249]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 10]
```diff
@@ -278,6 +324,10 @@ public boolean getBoolean(String columnName) {
         return (boolean) get(columnName);
     }
 
+    public Boolean getBooleanObject(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:327]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 11]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
         return (short) get(columnName);
     }
 
+    public short getShortObject(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:341]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 12]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
...
+    }
+
     @Override
     public int getInt(String columnName) {
         return (int) get(columnName);
     }
 
+    public int getInteger(String columnName) {
+        return (Integer) get(columnName);
+    }
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:352]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 13]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
...
+
     @Override
     public long getLong(String columnName) {
         return (long) get(columnName);
     }
 
+    public Long getLongObject(String columnName) {
+        return getLongObject(findColumn(columnName));
+    }
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:361]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 14]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
...
+
     @Override
     public float getFloat(String columnName) {
         return (float) get(columnName);
     }
 
+    public Float getFloatObject(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:368]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 15]
```diff
@@ -288,30 +338,50 @@ public short getShort(String columnName) {
...
+        return getFloatObject(findColumn(columnName));
+    }
+
     @Override
     public double getDouble(String columnName) {
         return (double) get(columnName);
     }
 
+    public Double getDoubleObject(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:377]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 16]
```diff
@@ -388,6 +458,18 @@ public JSONObject getJSONObject(int i) {
         return (JSONObject) get(i);
     }
 
+    public UUID getUUID(int i) {
+        Object u = get(i);
+        if (u == null) {
+            return null;
+        }
+        return UUID.fromString((String) u);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:466]: This part isn't tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 17]
```diff
@@ -388,6 +458,18 @@ public JSONObject getJSONObject(int i) {
...
+    }
+
+    public UUID getUUID(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:469]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 18]
```diff
@@ -862,12 +958,24 @@ public Date getDate(String columnName, Calendar calendar) throws SQLException {
...
     @Override
     public Time getTime(int i, Calendar calendar) throws SQLException {
-        return new Time(getDate(i, calendar).getTime());
+        return getTime(i);
     }
 
     @Override
     public Time getTime(String columnName, Calendar calendar) throws SQLException {
-        return getTime(findColumn(columnName), calendar);
+        return getTime(columnName);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:966]: Not tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 19]
```diff
@@ -862,12 +958,24 @@ public Date getDate(String columnName, Calendar calendar) throws SQLException {
...
+    }
+
+    public LocalTime getLocalTime(int i) {
+        Object time = get(i);
+        if (time == null) {
+            return null;
+        }
+        return LocalTime.parse((String) time);
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:974]: This part isn't tested

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 20]
```diff
@@ -862,12 +958,24 @@ public Date getDate(String columnName, Calendar calendar) throws SQLException {
...
+    }
+
+    public LocalTime getLocalTime(String columnName) {
```
Comment (yihezkel) [inline data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:977]: Not tested

File: data/src/test/java/com/microsoft/azure/kusto/data/ResultSetTest.java [hunk 21]
```diff
@@ -0,0 +1,93 @@
...
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.UUID;
+
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+public class ResultSetTest {
+    @Test
```
Comment (yihezkel) [inline data/src/test/java/com/microsoft/azure/kusto/data/ResultSetTest.java:20]: Ideally should be a separate test for each unit being tested.

If that would take too long, might be worth at least checking each return type with 1 unit test (e.g. input string value, input int value and input column name all in 1 unit test).

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 17 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:469]: it is

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 18 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:966]: it is

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 6 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:961]: Time is only hour, minute, second

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 16 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:466]: it is

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 15 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:377]: i dont want every method

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 3 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:292]: what

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 2 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:287]: i is index

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 1 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:268]: it works best

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 9 above)
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:249]: it is

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 22]
```diff
@@ -203,19 +216,40 @@ public long getLong(int i) {
         return (long) obj;
     }
 
+    public Long getLongObject(int i) {
+        Object obj = get(i);
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:220]: i is index

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 23]
```diff
@@ -189,11 +194,19 @@ public short getShort(int i) {
...
+        return (Short) get(i);
+    }
+
     @Override
     public int getInt(int i) {
         return (int) get(i);
     }
 
+    public Integer getInteger(int i) {
+        return (Integer) get(i);
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:207]: i is index

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java [hunk 24]
```diff
@@ -179,6 +180,10 @@ public boolean getBoolean(int i) {
         return (boolean) get(i);
     }
 
+    public Boolean getBooleanObject(int i) {
+        return (Boolean) get(i);
```
[the remark this answers is not in the document]
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:184]: dont understand

File: data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java (hunk 23 above)
[the remark this answers is not in the document]
Comment (ohadbitt) [reply data/src/main/java/com/microsoft/azure/kusto/data/KustoResultSetTable.java:215]: same - i is index

File: data/src/test/java/com/microsoft/azure/kusto/data/ResultSetTest.java (hunk 21 above)
Comment (ohadbitt) [reply data/src/test/java/com/microsoft/azure/kusto/data/ResultSetTest.java:18]: as talked - i dont think i should do this here as its too much code

**Review by yihezkel — APPROVED** [review]

**Pull request merged**